fhir2dataset/graphquery.py

```diff
@@ -244,36 +228,44 @@ def _where(self, **wheres):
                     "prefix": prefix,
                     "value": value,
                 }
-                resource_type = self.resources_alias_info[resource_alias]["resource_type"]
-                searchparam_to_element = self.fhir_rules.resourcetype_searchparam_to_element(
-                    resource_type=resource_type, search_param=search_param,
-                )
-                if searchparam_to_element:
-                    element = searchparam_to_element
-                else:
-                    element = search_param
-                # print(f"element: {element}")
-                modifier = element.split(":")[-1]
-                if modifier in MODIFIERS_POSS:
-                    element = ":".join(element.split(":")[:-1])
-                    # print(f"element modified: {element}")
-                self.resources_alias_info[resource_alias]["elements"]["where"].append(element)
+                element = self._check_searchparam_or_fhirpath(resource_alias, search_param)
+                # # print(f"element: {element}")
+                # modifier = element.split(":")[-1]
+                # if modifier in MODIFIERS_POSS:
+                #     element = ":".join(element.split(":")[:-1])
+                # print(f"element modified: {element}")
+                self.resources_alias_info[resource_alias]["elements"]["where"][
+                    search_param
+                ] = element
 
     @timing
     def _select(self, **selects):
         """updates the resources_alias_info attribute with the elements that must be retrieved for each alias
 
         Keyword Arguments:
-            **selects: the key is an alias, the value is a list containing expressions to leaf elements
+            **selects: the key is an alias, the value is a list containing fhirpath to leaf elements
         """  # noqa
         for resource_alias in selects.keys():
```

Comment:
can you loop on `select.items()` instead?

Reply:
Yes! it's changed